gcc complaints about mismatched types for format specifiers.

Need to determine if we should file an upstream bug.

ALso disables GCC 7 format-nonliteral error.

--- glib-2.44.1/gio/gdbusauthmechanismsha1.c	Fri Dec 19 13:49:48 2014
+++ glib-2.44.1/gio/gdbusauthmechanismsha1.c	Wed May 20 10:56:38 2015
@@ -272,7 +272,7 @@
                            G_IO_ERROR_FAILED,
                            _("Permissions on directory '%s' are malformed. Expected mode 0700, got 0%o"),
                            path,
-                           statbuf.st_mode & 0777);
+                           (unsigned int) statbuf.st_mode & 0777);
               g_free (path);
               path = NULL;
               goto out;
--- glib-2.44.1/gio/gdbusmessage.c	Tue May 12 19:39:04 2015
+++ glib-2.44.1/gio/gdbusmessage.c	Wed May 20 10:56:38 2015
@@ -3472,9 +3472,9 @@
               if (fstat (fds[n], &statbuf) == 0)
                 {
                   g_string_append_printf (fs, "%s" "dev=%d:%d", fs->len > 0 ? "," : "",
-                                          major (statbuf.st_dev), minor (statbuf.st_dev));
+                                          (int) major (statbuf.st_dev), (int) minor (statbuf.st_dev));
                   g_string_append_printf (fs, "%s" "mode=0%o", fs->len > 0 ? "," : "",
-                                          statbuf.st_mode);
+                                          (unsigned int) statbuf.st_mode);
                   g_string_append_printf (fs, "%s" "ino=%" G_GUINT64_FORMAT, fs->len > 0 ? "," : "",
                                           (guint64) statbuf.st_ino);
                   g_string_append_printf (fs, "%s" "uid=%u", fs->len > 0 ? "," : "",
@@ -3482,7 +3482,7 @@
                   g_string_append_printf (fs, "%s" "gid=%u", fs->len > 0 ? "," : "",
                                           (guint) statbuf.st_gid);
                   g_string_append_printf (fs, "%s" "rdev=%d:%d", fs->len > 0 ? "," : "",
-                                          major (statbuf.st_rdev), minor (statbuf.st_rdev));
+                                          (int) major (statbuf.st_rdev), (int) minor (statbuf.st_rdev));
                   g_string_append_printf (fs, "%s" "size=%" G_GUINT64_FORMAT, fs->len > 0 ? "," : "",
                                           (guint64) statbuf.st_size);
                   g_string_append_printf (fs, "%s" "atime=%" G_GUINT64_FORMAT, fs->len > 0 ? "," : "",
--- glib-2.52.0/gio/gunixconnection.c
+++ glib-2.52.0/gio/gunixconnection.c
@@ -162,11 +162,13 @@
     {
       gint i;
 
+#pragma GCC diagnostic warning "-Wformat-nonliteral"
       g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
         ngettext("Expecting 1 control message, got %d",
                  "Expecting 1 control message, got %d",
                  nscm),
         nscm);
+#pragma GCC diagnostic pop
 
       for (i = 0; i < nscm; i++)
         g_object_unref (scms[i]);
@@ -196,11 +198,13 @@
     {
       gint i;
 
+#pragma GCC diagnostic warning "-Wformat-nonliteral"
       g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
                    ngettext("Expecting one fd, but got %d\n",
                             "Expecting one fd, but got %d\n",
                             nfd),
                    nfd);
+#pragma GCC diagnostic pop
 
       for (i = 0; i < nfd; i++)
         close (fds[i]);
@@ -555,6 +559,7 @@
     {
       if (nscm != 1)
         {
+#pragma GCC diagnostic warning "-Wformat-nonliteral"
           g_set_error (error,
                        G_IO_ERROR,
                        G_IO_ERROR_FAILED,
@@ -562,6 +567,7 @@
                                 "Expecting 1 control message, got %d",
                                 nscm),
                        nscm);
+#pragma GCC diagnostic pop
           goto out;
         }
 
